### PR TITLE
CI: Call agent shutdown test

### PR DESCRIPTION
**Pull request by jodh-intel**
Run the agent shutdown test as part of CI testing code in this repo.

Fixes: #1808.

Depends-on:github.com/kata-containers/tests#3495

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by fidencio** [comment]
@jodh-intel, after checking the errors in different CIs, I'd say we should:
* ensure we have runc installed as when running this test;
* ensure we skip this test if not using qemu as the default hypervisor;

Once those two happen, we will see a better rate of "green" in the CI results.

**Comment by jodh-intel** [comment]
Hi @GabyCT - Something odd is happening with this PR. It's using the depends-on to call the script on https://github.com/kata-containers/tests/pull/3495. However, if you compare the Ubuntu 18.04 and 20.04 logs, they are running different versions of the `tracing/test-agent-shutdown.sh` script on that PR:

| Distro version | `ss` command run |
|-|-|
| Ubuntu 18.04 | `ss -HO --vsock` |
| Ubuntu 20.04 | `ss -H --vsock` |

Am I going mad? :smile:

/cc @chavafg

**Comment by GabyCT** [comment]
@jodh-intel the ubuntu 18.04 CI has been disable could you please test again and see if it is happening...right now only ubuntu 20.04 is running

**Comment by chavafg** [comment]
seems that the report you are getting from[ ubuntu-18.04 ](http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-PR/1472/)is 13 days old, so yes disregard this one. You can get rid of this report by repushing. 

The initrd jobs[1] seems to still running in 18.04 and looks like it reports the same as 20.04 job[2]

[1] http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-initrd-PR/1068/consoleFull
[2] http://jenkins.katacontainers.io/job/kata-containers-2.0-ubuntu-20.04-PR/96/consoleFull

**Comment by jodh-intel** [comment]
Thanks @GabyCT and @chavafg!

/test-ubuntu

**Comment by jodh-intel** [comment]
here we go again...

/test-ubuntu

**Comment by jodh-intel** [comment]
```
tracing/test-agent-shutdown.sh: line 540: 13409 Segmentation fault      (core dumped) agent_addr=$(get_agent_vsock_address || true)
```

Errr... wow. Do I win a prize? :smile:

**Comment by jodh-intel** [comment]
Metrics CI is failing, but not due to this test:

```
17:09:32 INFO: Install image to //usr/share/kata-containers
17:09:32 install: cannot remove '//usr/share/kata-containers/kata-containers.img': Device or resource busy
17:09:32 Failed at 61: sudo install -D "${file_to_install}" "${image_path}"
```

**Review by fidencio — APPROVED** [review]
lgtm, thanks @jodh-intel!

File: ci/run.sh [hunk 1]
```diff
@@ -13,4 +13,5 @@ clone_tests_repo
 
 pushd ${tests_repo_dir}
 .ci/run.sh
+tracing/test-agent-shutdown.sh
```
Comment (Jakob-Naucke) [inline ci/run.sh:16]: OOI: This is for the GHA checks only, right? Because it would be great to test on all architectures.

**Review by Jakob-Naucke — APPROVED** [review]

**Review by jcvenegas — APPROVED** [review]

**Comment by Jakob-Naucke** [comment]
@jodh-intel I know I approved this two days ago, but I just went through some mid-sized headache trying to find out why https://github.com/kata-containers/tests/pull/3813, which installs Rust via Makefile, wasn't being picked up. Is there any reason we don't run this through the tests repo, given especially that previously, the `kata-containers/ci/run.sh` script only ran `tests/.ci/run.sh` and nothing else?

**Comment by jodh-intel** [comment]
@Jakob-Naucke - Sorry about that. I did consider calling the test from the tests repo, but decided it might be best to call it from here instead since this is where the agent shutdown code lives. Hence, if we need to change that logic (and I think we might: see #2443), it would make the changes simpler to understand since the test could also be updated on the same PR.

**Pull request merged**